src/org/skypro/skyshop/product/SearchEngine.java

```diff
@@ -2,26 +2,23 @@
 
 import org.skypro.skyshop.exceptions.BestResultNotFound;
 
+import java.util.ArrayList;
+
 public class SearchEngine {
-    private final Searchable[] searchables;
+    private final ArrayList<Searchable> searchables;
     private int size;
 
     public SearchEngine() {
-        this.searchables = new Searchable[10];
-        size = 0;
+        searchables = new ArrayList<>();
     }
 
-    public Searchable[] searchByKeyword(String keyword) {
-        Searchable[] searchResult = new Searchable[5];
-        int count = 0;
+    public ArrayList<Searchable> searchByKeyword(String keyword) {
```

Comment:
тип должен быть максимально общим, чтобы использовать преимущества полиморфизма
```suggestion
    public List<Searchable> searchByKeyword(String keyword) {
```